runtime/lua/vim/snippet.lua

```diff
@@ -278,13 +278,26 @@ local function setup_autocmds(bufnr)
     desc = 'Update snippet state when the cursor moves',
     buffer = bufnr,
     callback = function()
+      local cursor_row, cursor_col = cursor_pos()
+
+      -- The cursor left the snippet region.
+      local snippet_range = get_extmark_range(bufnr, M._session.extmark_id)
+      if
+        cursor_row < snippet_range[1]
+        or (cursor_row == snippet_range[1] and cursor_col < snippet_range[2])
+        or cursor_row > snippet_range[3]
+        or (cursor_row == snippet_range[3] and cursor_col > snippet_range[4])
+      then
+        M.exit()
```

Comment:
emm problem is when jump to second section , before section still has highlight. are you sure there invoke `exit` is correct.

Reply:
You mean jumping to the second tabstop? As long as you don't leave the inserted snippet text, the highlights should remain.

I'm wary of modifying this further, since different users might want the snippet to be cancelled at different moments.

Reply:
prefer something like 
```diff
diff --git a/runtime/lua/vim/snippet.lua b/runtime/lua/vim/snippet.lua
index 7680d2e..50ab080 100644
--- a/runtime/lua/vim/snippet.lua
+++ b/runtime/lua/vim/snippet.lua
@@ -504,6 +504,8 @@ end
 ---
 --- @param direction (vim.snippet.Direction) Navigation direction. -1 for previous, 1 for next.
 function M.jump(direction)
+  local line = vim.api.nvim_win_get_cursor(0)[1]
+  vim.api.nvim_buf_clear_namespace(0, snippet_ns, line, line + 1)
   -- Get the tabstop index to jump to.
   local dest_index = M._session and M._session:get_dest_index(direction)
   if not dest_index then
```

Reply:
The highlights are applied to the tabstops of an active snippet. I would like other people to weigh in on how customizable the highlights should be before I make any more changes.

Reply:
emm I mean when you jump to a new section like $2. The active $2 . The previous highlight should not be retained in $1. This has nothing to do with customization I guess.

Reply:
> when jump to next or before tabstop. before tabstop highlight should remove not keep.

No. As long as you _can_ jump to these tabstops, they should remain to be highlighted.

Reply:
Also, this discussion is unrelated to the current PR, which fixes a specific bug.

Reply:
what i mean stop edit back to normal . they are highlight. and can't cancel highlight ? not bug ???

Reply:
fair. I encountered it adding background highlighting to my entire buffer. It's just that I can't find the steps to reproduce it 100%.

Reply:
Well, then you should have lead with that. That's a bug that needs to be fixed, and not by disabling functionality.

Reply:
still confused. Why keep the before or after highlighted. I had to add the clear namespace in the keymap. my suggest remove `highlight` we should keep things simple. anyone want do it locally.

Reply:
> this discussion is unrelated to the current PR, which is about exiting the snippet correctly.

I echo @clason here. Resolving this as the discussion about snippets highlights should be handled as its own feature request.